packages/headless/src/features/relative-date-set/relative-date-set-state.ts

```diff
@@ -0,0 +1,7 @@
+import {RelativeDate} from './relative-date';
+
+export type RelativeDateSetState = Record<string, Record<string, RelativeDate>>;
```

Comment:
Why do we need a new reducer and state? Would it be possible to register date ranges from relative dates instead?

Reply:
What you're proposing is similar to the original solution: https://github.com/coveo/ui-kit/pull/973